### PR TITLE
feat: add support for verkle

### PR DESCRIPTION
**Pull request by barnabasbusa**

### CODE REVIEW

File: package_io/input_parser.star [hunk 1]
```diff
@@ -96,6 +96,12 @@ def parse_input(input_args):
 	if result["network_params"]["deneb_fork_epoch"] == 0:
 		fail("deneb_fork_epoch is 0 needs to be > 0 ")
 
+	if result["network_params"]["electra_fork_epoch"] != None and result["network_params"]["deneb_fork_epoch"] < result["network_params"]["electra_fork_epoch"]:
```
Comment (h4ck3rk3y) [inline package_io/input_parser.star:99]: can deneb and electra be the same say 5,5 - or should we fail that as well?

File: static_files/genesis-generation-config/el/genesis-config.yaml.tmpl [hunk 2]
```diff
@@ -18,3 +18,4 @@ genesis_delay: {{ .GenesisDelay }}
 slot_duration_in_seconds: {{ .SecondsPerSlot }}
 capella_fork_epoch: {{ .CapellaForkEpoch }}
 deneb_fork_epoch: {{ .DenebForkEpoch }}
+electra_fork_epoch: {{ .ElectraForkEpoch }}
```
Comment (h4ck3rk3y) [inline static_files/genesis-generation-config/el/genesis-config.yaml.tmpl:21]: How does this handle none values?

**Review by h4ck3rk3y — APPROVED** [review]

File: package_io/input_parser.star (hunk 1 above)
Comment (barnabasbusa) [reply package_io/input_parser.star]: So after some discussion we agreed that deneb can't happen when electra is defined. Adjusted accordingly

File: static_files/genesis-generation-config/el/genesis-config.yaml.tmpl (hunk 2 above)
Comment (barnabasbusa) [reply static_files/genesis-generation-config/el/genesis-config.yaml.tmpl:21]: Unknown values are ignored by CLs

File: package_io/input_parser.star [hunk 3]
```diff
@@ -96,6 +96,13 @@ def parse_input(input_args):
 	if result["network_params"]["deneb_fork_epoch"] == 0:
 		fail("deneb_fork_epoch is 0 needs to be > 0 ")
 
+	if result["network_params"]["electra_fork_epoch"] != None:
+		# if electra is defined, then deneb needs to be set very high
+		result["network_params"]["deneb_fork_epoch"] = 200000
```
Comment (h4ck3rk3y) [inline package_io/input_parser.star:101]: lets take out the constant 20000 and put it in a value up top like HIGH_DENEB_VALUE_FORK_VERKLE = 20000

File: src/el/el_client_context.star [hunk 4]
```diff
@@ -7,7 +7,8 @@ def new_el_client_context(
 	ws_port_num,
 	engine_rpc_port_num,
 	jwt_secret,
-	service_name = ""):
+	service_name = "",
+	electra_fork_epoch = None):
```
Comment (h4ck3rk3y) [inline src/el/el_client_context.star:11]: I don't think you need this here in el_client_context!

**Review by h4ck3rk3y — APPROVED** [review]

File: package_io/input_parser.star (hunk 3 above)
Comment (barnabasbusa) [reply package_io/input_parser.star:101]: ok

File: src/el/el_client_context.star (hunk 4 above)
Comment (barnabasbusa) [reply src/el/el_client_context.star]: you are right

**Pull request merged**